### PR TITLE
KAZOO-4295: Add new error code 423/locked

### PR DESCRIPTION
**Pull request by macpie**

### CODE REVIEW

File: applications/crossbar/src/modules/cb_token_auth.erl [hunk 1]
```diff
@@ -209,10 +209,15 @@ is_expired(Context, JObj) ->
     AccountId = wh_json:get_value(<<"account_id">>, JObj),
     case wh_util:is_account_expired(AccountId) of
         'false' -> check_as(Context, JObj);
-        'true' ->
+        {'true', Expired} ->
             _ = wh_util:spawn(fun() -> maybe_disable_account(AccountId) end),
-            Cause = wh_json:from_list([{<<"cause">>, <<"account expired">>}]),
-            {'halt', cb_context:add_system_error('forbidden', Cause, Context)}
+            Cause =
+                wh_json:from_list([
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_token_auth.erl]: Drop the `[` to a new line. Should be:

```erlang
wh_json:from_list(
    [{k, v}
     ,{k2, v2}
    ]
)
```

File: applications/crossbar/src/modules/cb_user_auth.erl [hunk 2]
```diff
@@ -260,8 +260,11 @@ maybe_account_is_enabled(Context, Credentials, Method, Account) ->
             maybe_authenticate_user(Context, Credentials, Method, Account);
         'false' ->
             lager:debug("account ~p is disabled", [Account]),
-            Props = [{'details', <<"account_disabled">>}],
-            cb_context:add_system_error('forbidden', wh_json:from_list(Props), Context)
+            Cause =
+                wh_json:from_list([
+                    {<<"message">>, <<"account disabled">>}
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_user_auth.erl]: Same here, or put it all on one line.

File: applications/crossbar/src/modules/cb_token_auth.erl [hunk 3]
```diff
@@ -209,10 +209,16 @@ is_expired(Context, JObj) ->
     AccountId = wh_json:get_value(<<"account_id">>, JObj),
     case wh_util:is_account_expired(AccountId) of
         'false' -> check_as(Context, JObj);
-        'true' ->
+        {'true', Expired} ->
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_token_auth.erl]: Did you forget to commit a change to `wh_util`? `wh_util:is_account_expired/1` returns a `boolean()` only. I hope that's the case versus this being an untested changeset!

Comment (macpie) [inline applications/crossbar/src/modules/cb_token_auth.erl]: No just forgot to publish that file

File: core/whistle-1.0.0/src/wh_util.erl [hunk 4]
```diff
@@ -401,17 +401,20 @@ is_account_enabled(Account) ->
 
     end.
 
--spec is_account_expired(api_binary()) -> boolean().
+-spec is_account_expired(api_binary()) -> 'false' | {'true', integer()}.
```
Comment (jamesaimonetti) [inline core/whistle-1.0.0/src/wh_util.erl]: use `gregorian_seconds()` instead to indicate the type of integer this represents.

**Comment by jamesaimonetti** [comment]
Merged manually

**Pull request closed without merging**